private/model/api/operation.go

```diff
@@ -139,8 +139,27 @@ func (c *{{ .API.StructName }}) {{ .ExportedName }}Request(` +
 func (c *{{ .API.StructName }}) {{ .ExportedName }}(` +
 	`input {{ .InputRef.GoType }}) ({{ .OutputRef.GoType }}, error) {
 	req, out := c.{{ .ExportedName }}Request(input)
-	err := req.Send()
-	return out, err
+	return out, req.Send()
+}
+
+// {{ .ExportedName }}WithContext is the same as {{ .ExportedName }} with the addition of
+// the ability to pass a context and additional request options.
+//
+// See {{ .ExportedName }} for details on how to use this API operation.
+//
+// The context will be used to cancel the request, and any retry delays. The
+// underlying request's retries may be given sub contexts in the future.
+// See the aws/request package's Request and Option types for configurations.
+//
+// The Context passed in must not be nil or a panic may occur.
```

Comment:
Since there's a chance of a panic (and the caller is forbidden from passing a nil context), it may be better to 1) say "The Context provided must not be nil", and 2) for aws.request.Request.SetContext to explicitly check for nil and to panic (as net/http.Request.WithContext does). When the panic is reliable the user will learn how to use the API very quickly, rather than "getting lucky" for a while and only encountering the panic once their app is deployed.

(And same for `*PagesWithContext` and `WaitUntil*WithContext`)